### PR TITLE
Refactoring and cleanup of torch-ddp

### PR DESCRIPTION
**Pull request by charlesfrye**
This PR factors out the boilerplate code from the two scripts and then merges them into one script, with the logging behavior changed by a flag, `--log_all`.

This reduces the amount of repeated code substantially.

I tested this branch only on a single-GPU setup ([Colab](https://colab.research.google.com/drive/1bvPIe6IG0AmDPfsTWtnwczC5qPlSOzZh?usp=sharing)), so I can't confirm that it works on a multi-GPU machine. @ariG23498 how/have you been testing on multi-GPU?

### CODE REVIEW

**Comment by ayulockin** [comment]
Hey @charlesfrye, @ariG23498 asked me to test the scripts since I have access to multi-gpu. 

Method 1 works as intended. Here's the [W&B run page](https://wandb.ai/ayush-thakur/torch-ddp/runs/1waa6gyv?workspace=user-ayush-thakur).

Method 2 is giving this error.

image

File: examples/pytorch/pytorch-ddp/README.md [hunk 1]
```diff
@@ -31,22 +37,27 @@ python -m torch.distributed.launch \
...
-2. Call `wandb.log()` in any process where you want to log metrics. This is ok because all processes have called `wandb.init()`, so they can call other `wandb` functions safely.
-
-
-    Usage: 
-    ```python
-    python -m torch.distributed.launch \
-      --nproc_per_node 2 \
-      --nnodes 1 \
-      --node_rank 0 \
-      log-all.py \
```
Comment (ayulockin) [inline examples/pytorch/pytorch-ddp/README.md:47]: Here's a typo. It should be `--log_all`.

Comment (ayulockin) [reply examples/pytorch/pytorch-ddp/README.md:47]: Here's the result after correcting the typo:

image

Comment (ariG23498) [reply examples/pytorch/pytorch-ddp/README.md:47]: LGTM ⭐ 
Thanks for the review!

**Pull request merged**